lib/src/svg/xml_parsers.dart

```diff
@@ -157,27 +207,39 @@ PaintServer parseRadialGradient(XmlElement el) {
   final List<double> offsets = new List<double>(stops.length);
   parseStops(stops, colors, offsets);
 
+  final Matrix4 originalTransform = parseTransform(getAttribute(el, 'gradientTransform', def: null));
+
   return (Rect bounds) {
-    final double cx = _parseDecimalOrPercentage(
-      rawCx,
-      multiplier: bounds.width + bounds.left + bounds.left,
-    );
-    final double cy = _parseDecimalOrPercentage(
-      rawCy,
-      multiplier: bounds.height + bounds.top + bounds.top,
-    );
-    final double r = _parseDecimalOrPercentage(
-      getAttribute(el, 'r', def: '50%'),
-      multiplier: (bounds.width + bounds.height) / 2,
-    );
-    final double fx = _parseDecimalOrPercentage(
-      getAttribute(el, 'fx', def: rawCx),
-      multiplier: bounds.width + (bounds.left * 2),
-    );
-    final double fy = _parseDecimalOrPercentage(
-      getAttribute(el, 'fy', def: rawCy),
-      multiplier: bounds.height + (bounds.top),
-    );
+    double cx, cy, r, fx, fy;
+    Matrix4 transform = originalTransform?.clone() ?? new Matrix4.identity();
+
+    if (isObjectBoundingBox) {
+      final Matrix4 scale = affineMatrix(bounds.width, 0.0, 0.0, bounds.height, 0.0, 0.0);
+      final Matrix4 translate = affineMatrix(1.0, 0.0, 0.0, 1.0, bounds.left, bounds.top);
+      transform = translate.multiplied(scale)..multiply(transform);
+
+      cx = _parseDecimalOrPercentage(rawCx);
+      cy = _parseDecimalOrPercentage(rawCy);
+      r = _parseDecimalOrPercentage(rawR);
+      fx = _parseDecimalOrPercentage(rawFx);
+      fy = _parseDecimalOrPercentage(rawFy);
+    } else {
+      if (_isPercentage(rawCx)
+        || _isPercentage(rawCy)
+        || _isPercentage(rawR)
+        || _isPercentage(rawFx)
+        || _isPercentage(rawFy)) {
+        // TODO: Support userSpaceOnUse with percentage values
```

Comment:
I'm pretty sure we should be able to just use `window.physicalSize` from `dart:ui`.  

I'd like to avoid the `NoopShader` - it seems dangerous and avoidable in this case.

Reply:
Ahhh, that's the confusion.

In that case, we could add another parameter to the gradient parsers, something like `parseLinearGradient(XmlElement el, Rect viewBox)`.  the `bounds` parameter passed into `parseSvgElement` and `parseSvgGroup` is actually the `viewBox`.  That could be passed through to `parseDefs`, which could pass it to `parseGradients`, which could pass it to `parse[Linear|Radial]Gradient`.

Does that make sense to you?

Reply:
there are actually 2 `viewBoxes`:
1) the one of the SVG root element
2) the one of the object that is being drawn

When dealing with gradients, user can describe them either by 1) (`userSpaceOnUse`) or by 2)(`objectBoundingBox`). It looks like (but I might be wrong) that the `bounds` parameter refers to the inner object (e.g. the Rect or the Path node) w.r.t the root viewBox that is described as a Rect with origin 0,0 and size of the SVG root node.

Assuming that this is the case, the value that we are missing in order to be able to support the last case would be the SVG root node bounds, which we should either inject all the way down in every `parse` method or maybe make available inside every `XmlElement` as the `root` node.

Reply:
I think I understand you - what I'm saying is that you can pass in that root viewBox bounds like so:

1. Parse the viewBox from the SVG root 
2. Pass that viewBox as the bounds parameter to `parseSvgElement`
3. Pass that viewBox as a new parameter to `parseDefs`
4. Pass from item 3 to `parseGradient`
5. Pass from item 4 to `parseLinearGradient` and `parseRadialGradient`

1 and 2 are done.  3 to 5 could be added in this patch.

The inner functions returned in `parseLinearGradient` and `parseRadialGradient` should not be modified - they're going to continue to need the bounds they get today.  However, the outter method can take the viewBox in and use that to figure out the `userSpaceOnUse` percentages if necessary.

Reply:
ok, this makes sense! I'm going to try to patch the code and ping you back if I find some issues :)